node/pkg/watchers/sui/watcher.go

```diff
@@ -192,13 +192,53 @@ func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
 		return errors.New("type mismatch")
 	}
 
-	fields := *body.Fields
-	if (fields.ConsistencyLevel == nil) || (fields.Nonce == nil) || (fields.Payload == nil) || (fields.Sender == nil) || (fields.Sequence == nil) {
+	fieldsMap := *body.Fields
+	fm_cl, clExists := fieldsMap["consistency_level"]
+	fm_n, nonceExists := fieldsMap["nonce"]
+	fm_pl, payloadExists := fieldsMap["payload"]
+	fm_sndr, senderExists := fieldsMap["sender"]
+	fm_seq, sequenceExists := fieldsMap["sequence"]
+	fm_ts, timestampExists := fieldsMap["timestamp"]
+
+	// Check if all required fields exist
+	if (!clExists) || (!nonceExists) || (!payloadExists) || (!senderExists) || (!sequenceExists) || (!timestampExists) {
 		logger.Info("Missing required fields in event.")
```

Comment:
should we include more info about what’s missing?

Reply:
Now that you mention it, I don't really know that it should have any of these fields.  It could be a non-wormhole event.  The real question is how do I know this is a WH event and needs these fields vs. this is not a WH event at all?

Reply:
Dumping out the Map.